Compiling IRremoteESP8266 even if it should not, and it fails on ESP32 (heatpumpir)

### The problem

Library IRremoteESP8266 has been "removed" from heatpumpir for ESP32, but ESPHome still tries to compile it, and it fails.
After manually removing /root/.platformio/lib/IRremoteESP8266/ in the container, it works as expected.

### Which version of ESPHome has the issue?

2025.7.4

### What type of installation are you using?

Docker

### What platform are you using?

ESP32

### Component causing the issue

heatpumpir

### YAML Config

```yaml
esphome:
  name: stue

esp32:
  board: esp32dev
  framework:
    type: arduino

# Enable logging
logger:

web_server:

# MQTT
mqtt:
  broker: !secret mqtt_broker
  id: mqtt_client
  
light:
  - platform: status_led
    id: status
    name: "Status"
    pin: GPIO2

ota:
  platform: esphome
  password: ""

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: "Stue Fallback Hotspot"
    password: "QdYVbVDjwknW"

captive_portal:

remote_transmitter:
  pin: 
    number: GPIO23
    inverted: false
  carrier_duty_percent: 50%

climate:
  - platform: heatpumpir
    id: heatpump
    protocol: mitsubishi_fd
    name: "Stue"
    horizontal_default: "auto"
    vertical_default: "auto"
    max_temperature: 31
    min_temperature: 16
    sensor: room_temp

sensor:
  - platform: uptime
    name: Uptime Sensor
  - platform: internal_temperature
    name: "Internal Temperature"
  - platform: mqtt_subscribe
    internal: true
    unit_of_measurement: "°C"
    id: room_temp
    topic: ha_states/sensor/romtemperatur_stue/state
      
text_sensor:
  - platform: version
    name: "ESPHome Version"
```

### Anything in the logs that might be useful for us?

```txt
Archiving .pioenvs/stue/libdcd/libUpdate.a
Compiling .pioenvs/stue/libc16/ESPAsyncWebServer/AsyncWebSocket.cpp.o
Compiling .pioenvs/stue/libc16/ESPAsyncWebServer/BackPort_SHA1Builder.cpp.o
Compiling .pioenvs/stue/libc16/ESPAsyncWebServer/ChunkPrint.cpp.o
Compiling .pioenvs/stue/libc16/ESPAsyncWebServer/Middleware.cpp.o
Compiling .pioenvs/stue/libc16/ESPAsyncWebServer/WebAuthentication.cpp.o
Compiling .pioenvs/stue/libc16/ESPAsyncWebServer/WebHandlers.cpp.o
Compiling .pioenvs/stue/libc16/ESPAsyncWebServer/WebRequest.cpp.o
Compiling .pioenvs/stue/libc16/ESPAsyncWebServer/WebResponses.cpp.o
Compiling .pioenvs/stue/libc16/ESPAsyncWebServer/WebServer.cpp.o
Compiling .pioenvs/stue/libed7/AsyncUDP/AsyncUDP.cpp.o
Compiling .pioenvs/stue/lib40b/DNSServer/DNSServer.cpp.o
Archiving .pioenvs/stue/libed7/libAsyncUDP.a
Compiling .pioenvs/stue/lib634/ESPmDNS/ESPmDNS.cpp.o
Compiling .pioenvs/stue/lib321/IRremoteESP8266/IRac.cpp.o
Archiving .pioenvs/stue/libc16/libESPAsyncWebServer.a
Compiling .pioenvs/stue/lib321/IRremoteESP8266/IRrecv.cpp.o
Archiving .pioenvs/stue/lib634/libESPmDNS.a
Compiling .pioenvs/stue/lib321/IRremoteESP8266/IRsend.cpp.o
/root/.platformio/lib/IRremoteESP8266/src/IRrecv.cpp: In function 'void gpio_intr()':
/root/.platformio/lib/IRremoteESP8266/src/IRrecv.cpp:246:3: error: 'timerAlarmEnable' was not declared in this scope; did you mean 'timerAlarm'?
  246 |   timerAlarmEnable(timer);
      |   ^~~~~~~~~~~~~~~~
      |   timerAlarm
/root/.platformio/lib/IRremoteESP8266/src/IRrecv.cpp: In member function 'void IRrecv::enableIRIn(bool)':
/root/.platformio/lib/IRremoteESP8266/src/IRrecv.cpp:362:21: error: too many arguments to function 'hw_timer_t* timerBegin(uint32_t)'
  362 |   timer = timerBegin(_timer_num, 80, true);
      |           ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
In file included from /config/.esphome/platformio/packages/framework-arduinoespressif32@src-702d0f93023d86e22d8ef62aa333f0b7/cores/esp32/esp32-hal.h:98,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif32@src-702d0f93023d86e22d8ef62aa333f0b7/cores/esp32/Arduino.h:45,
                 from /root/.platformio/lib/IRremoteESP8266/src/IRrecv.h:10,
                 from /root/.platformio/lib/IRremoteESP8266/src/IRrecv.cpp:6:
/config/.esphome/platformio/packages/framework-arduinoespressif32@src-702d0f93023d86e22d8ef62aa333f0b7/cores/esp32/esp32-hal-timer.h:35:13: note: declared here
   35 | hw_timer_t *timerBegin(uint32_t frequency);
      |             ^~~~~~~~~~
/root/.platformio/lib/IRremoteESP8266/src/IRrecv.cpp:371:3: error: 'timerAlarmWrite' was not declared in this scope; did you mean 'timerWrite'?
  371 |   timerAlarmWrite(timer, MS_TO_USEC(params.timeout), ONCE);
      |   ^~~~~~~~~~~~~~~
      |   timerWrite
/root/.platformio/lib/IRremoteESP8266/src/IRrecv.cpp:375:23: error: too many arguments to function 'void timerAttachInterrupt(hw_timer_t*, void (*)())'
  375 |   timerAttachInterrupt(timer, &read_timeout, false);
      |   ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/config/.esphome/platformio/packages/framework-arduinoespressif32@src-702d0f93023d86e22d8ef62aa333f0b7/cores/esp32/esp32-hal-timer.h:50:6: note: declared here
   50 | void timerAttachInterrupt(hw_timer_t *timer, void (*userFunc)(void));
      |      ^~~~~~~~~~~~~~~~~~~~
/root/.platformio/lib/IRremoteESP8266/src/IRrecv.cpp: In member function 'void IRrecv::disableIRIn()':
/root/.platformio/lib/IRremoteESP8266/src/IRrecv.cpp:401:3: error: 'timerAlarmDisable' was not declared in this scope
  401 |   timerAlarmDisable(timer);
      |   ^~~~~~~~~~~~~~~~~
/root/.platformio/lib/IRremoteESP8266/src/IRrecv.cpp: In member function 'void IRrecv::resume()':
/root/.platformio/lib/IRremoteESP8266/src/IRrecv.cpp:429:3: error: 'timerAlarmDisable' was not declared in this scope
  429 |   timerAlarmDisable(timer);
      |   ^~~~~~~~~~~~~~~~~
Archiving .pioenvs/stue/lib40b/libDNSServer.a
Compiling .pioenvs/stue/lib321/IRremoteESP8266/IRtext.cpp.o
*** [.pioenvs/stue/lib321/IRremoteESP8266/IRrecv.cpp.o] Error 1
========================= [FAILED] Took 97.13 seconds =========================
```

### Additional information

_No response_

--- a/esphome/components/heatpumpir/climate.py
+++ b/esphome/components/heatpumpir/climate.py
@@ -127,5 +127,5 @@ async def to_code(config):
     cg.add(var.set_min_temperature(config[CONF_MIN_TEMPERATURE]))
 
     cg.add_library("tonia/HeatpumpIR", "1.0.37")
-    if CORE.is_libretiny:
+    if CORE.is_libretiny or CORE.is_esp32:
         CORE.add_platformio_option("lib_ignore", "IRremoteESP8266")

